src/components/playground/Comment.jsx

```diff
@@ -0,0 +1,193 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import Radium from 'radium';
+
+import moment from 'moment';
+
+import ReactCSSTransitionGroup from 'react-addons-css-transition-group';
+
+import ProfileBadge from './ProfileBadge';
+import ProfileInfo from './ProfileInfo';
+import CommentContent from './CommentContent';
+import ActionsBar from './ActionsBar';
+import Upvoter from './Upvoter';
+import CommentTools from './CommentTools';
+
+import FaCog from 'react-icons/lib/fa/cog';
+import CoralIcon from '../../components/CoralIcon';
+
+import mediaQueries from '../../playgroundSettings';
+
+@connect(state => state.playground)
+@Radium
+class Comment extends React.Component {
+
+  onToolsClick() {
+    this.setState({ toolsExpanded: !this.state.toolsExpanded });
+  }
+
+  onProfileClick() {
+    this.setState({ profileExpanded: !this.state.profileExpanded });
+  }
+
+  render() {
+
+    var commentTools = this.state.toolsExpanded ? <CommentTools /> : null;
+    var profileInfoSection = this.state.profileExpanded ? <ProfileInfo user={ this.props.user } /> : null;
+
+    var leftPadding = 0;
+    if (this.props.togglerGroups.layout.togglers.profilepictures.status) leftPadding += 75;
+    if (this.props.depth > 0) leftPadding += 25;
+
+    var user = this.props.users[this.props.user];
+
+    var pulsateTools = this.props.pulseAnimation && this.props.pulseTarget == "commentTools";
+    var pulsateName = this.props.pulseAnimation && this.props.pulseTarget == "commentName";
+
+    return (
+
+      <div style={ 
+        [ 
+          styles.comment,
+          { paddingLeft: leftPadding },
+          this.props.depth > 0 ? { borderLeft: '1px solid #ddd' } : null
+        ]
+      }>
+        <h4 style={ [ 
+              styles.userName, 
+              this.props.togglerGroups['reputation'].togglers['badges'].status && 
+                user.badges && 
+                user.badges.length ? 
+                  styles.withBadge 
+                : null,
+                pulsateName ? styles.colorPulse : null
+           ] } 
+           onClick={ this.onProfileClick.bind(this) }>
+
+          <ReactCSSTransitionGroup transitionName="fade" transitionAppear={ false }>
+          { 
+            this.props.togglerGroups['reputation'].togglers['badges'].status ? 
+              <span>
+                {
+                  user.badges && user.badges.length ? 
+                    <CoralIcon 
+                      style={ styles.badgeIcon } 
+                      size="medium" 
+                      name={ user.badges[0].icon } 
+                      color={ user.badges[0].color } />
+                  : null
+                }
+              </span>
+            : null
+          }
+          </ReactCSSTransitionGroup>
+          { 
+            this.props.togglerGroups['privacy'].togglers['anonymity'].status ? 
```

Comment:
I'm thinking I could encapsulate this into something more handy `getStatus('privacy', 'anonymity')`, also that `.status` property should probably be called `enabled`.